django/applications/catmaid/control/volume.py

```diff
@@ -1245,22 +1245,34 @@ def get_volume_data(project_id, volume_ids):
         mesh_name = r[name_idx]
         mesh_id = r[id_idx]
 
-        mesh_type = re.search('<(.*?) ', mesh_str).group(1)
+        mesh_match = re.search('<(.*?) ', mesh_str)
+        if mesh_match is None:
+            raise Exception('Malformed input: mesh type not found')
+        mesh_type = mesh_match.group(1)
 
         # Now reverse engineer the mesh
         if mesh_type == 'IndexedTriangleSet':
-            t = re.search("index='(.*?)'", mesh_str).group(1).split(' ')
-            faces = [(int(t[i]), int(t[i + 1]), int(t[i + 2]))
+            index_match = re.search("index='(.*?)'", mesh_str)
+            if index_match is None:
```

Comment:
It's pretty cool that `mypy` can catch situations like this in which the error case (`is None`) isn't handled properly (like before at this place).